### ISSUE TITLE
Multiplexing all the way through - edge (http/2 + http/3) -> envoy -> (tcp) server

### ISSUE THREAD

**Issue by vitruvvius** [envoyproxy/envoy#23994]
tl;dr **How does envoy multiplex into raw TCP connections from inbound http/2 & http/3 requests (if at all)?**

I've read numerous discussions on terminating HTTP/2 + HTTP/1 & re-writing bytes to raw TCP connections.
https://github.com/envoyproxy/envoy/pull/14552
Still confused *how* the envoy decap (http/2 -> tcp) multiplexes (or pipelines) requests into TCP clusters.
Comments reference pipelining, though multiplexing logically consistent.

Example:
Edge client sends 16 multiplexed requests (http/2 + http/3) -> envoy -> (tcp) server/cluster
How does envoy multiplex outbound to *raw* TCP (if at all) after receiving inbound http/2 request?
If it does multiplex, where does TCP cluster parse stream-id (to return a response)?
If not, how does envoy open/share/pool/pipeline TCP connections?
If a single connection, how are the 16 requests ordered (serial, time it hits the queue)?

Thanks for a wonderful tool.

**Comment by zuercher** [comment]
@alyssawilk maybe?

**Comment by vitruvvius** [comment]
Thanks @alyssawilk!

Great to hear. I'll note 16 / K connections opened into a pool, requests sent w/ out ordering.
